packages/react/src/primitives/PhoneNumberField/PhoneNumberField.tsx

```diff
@@ -6,42 +6,71 @@ import { CountryCodeSelect } from './CountryCodeSelect';
 import { PhoneNumberFieldProps, Primitive } from '../types';
 import { ComponentText } from '../shared/constants';
 import { TextField } from '../TextField';
+import { useDeprecationWarning } from '../../hooks/useDeprecationWarning';
 
 const PhoneNumberFieldPrimitive: Primitive<PhoneNumberFieldProps, 'input'> = (
   {
     autoComplete = 'tel-national',
     className,
     countryCodeName,
-    countryCodeLabel = ComponentText.PhoneNumberField.countryCodeLabel,
+    dialCodeName,
+    countryCodeLabel = ComponentText.PhoneNumberField.dialCodeLabel,
+    dialCodeLabel = ComponentText.PhoneNumberField.dialCodeLabel,
     defaultCountryCode,
+    defaultDialCode,
     hasError,
     isDisabled,
     isReadOnly,
     onCountryCodeChange,
+    onDialCodeChange,
     onInput,
     size,
     type,
     variation,
     countryCodeRef,
+    dialCodeRef,
     dialCodeList,
     ...rest
   },
   ref
 ) => {
+  /**
+   * Merge all dial/country code values in preparation of countryCode values being removed preferring dial code values
+   */
+  dialCodeName = dialCodeName || countryCodeName;
+  dialCodeLabel = dialCodeLabel || countryCodeLabel;
+  defaultDialCode = defaultDialCode || defaultCountryCode;
+  onDialCodeChange = onDialCodeChange || onCountryCodeChange;
+  dialCodeRef = dialCodeRef || countryCodeRef;
```

Comment:
Few things here: 
- are we documenting the logic here so users will know if for example, if both `dialCodeName` and `countryCodeName` that `dialCodeName` will "win" and that `countryCodeName` will be not be used?
- can we consider an approach that avoid mutating props?

Reply:
+1